### ISSUE TITLE
Logging plugin

### ISSUE THREAD

**Issue by iskrisis** [k-next/plugins#13]
I have implemented cruel logs before it is pretty handy on bigger sites. With some nice panel widget it can be pretty handy.

There was field https://github.com/texnixe/kirby-logger but there probably should be some helper functions.

Examples of usecases:
- Logging sending emails or submitting forms
- Logging when somebody visits error page
- Logging of things like created thumbs, backups etc

**Comment by bnomei** [comment]
This lib might be a bit over the top but well tested: https://github.com/Seldaek/monolog

**Comment by bnomei** [comment]
i would be willing to help on this plugin if @iskrisis needs a collaborator (then just assign me :D).

**Comment by bnomei** [comment]
would be awesome if this works part of the field method chain and pagemethod. like

```
$page->image('wow.jpg')->thumb()->log();
if($page->image('not-found.gif') == null) {
  $page->log()->error('image not found: ' + 'not-found.gif' ); // has $page to add to log
}

**Comment by iskrisis** [comment]
This was just idea. I currently don't have plans how to use it so it could took me quite some time.

Also i am not sure if i am experienced enough to decide the right api for something like this. 

@bnomei to your solution - i am not sure if it is good idea to tie to page object. I mean you could use this in many situations where you don't have $page object. 
Looking at: 
https://laravel.com/docs/5.6/logging 
or 
https://www.slimframework.com/docs/v2/logging/overview.html 

It seems to me logging is one of those things where globals make sense. Considering Kirby conventions i think api should be similar to Slim framework.

**Comment by bnomei** [comment]
agreed. it needs global functions. just thought the chaining would be neat to avoid passing around the objects triggering the log.

**Comment by bvdputte** [comment]
I had a go at it (more to figure out how composer etc works) and built a simple logger utility: https://github.com/bvdputte/kirby-log

**Comment by iskrisis** [comment]
Awesome.

Two things come to mind 
- I think most people will use only one log file maybe some main log set in config would be a good idea.
```
c::set('log', true);
c::set('log.name', 'mylog');
```

- Have you already implemented Klogs priority (->error(), ->critical()) etc? If so it calls for kirby-log-viewer field - maybe something i could have go at to also try making fields for kirby.

**Comment by bvdputte** [comment]
Yes, I thought about that too. Actually, if we would work with a "default" log, then we could create static function for it like `kirbylogger::write("message");`.

I haven't tampered with the levels (everything is hardcoded as "INFO" now), but this is something which should not be very hard to do I think.

**Comment by bnomei** [comment]
@bvdputte please consider the option configure the [format](https://github.com/katzgrau/KLogger#tab-separated). i would need this to make the created logs better consumable by other processes – like json or csv. readabilty of the raw log is not my major concern.

**Comment by iskrisis** [comment]
@bvdputte i agree with the static function but the problem is that in rather newbie Kirby world static functions are not that common. That might change with Kirby 3 maybe because of it being composer ready?

I see new changes. Actually what you have done api is more kirby like compared to static function.

Also maybe you don't have to namespace it like this ```$kirby->option("bvdputte.kirbylog.logfolder"```? Or is it something new we should do?

**Comment by bnomei** [comment]
@iskrisis afaik you have to namespace options. not when defining in you plugin config file, but when calling them somewhere else.

**Comment by bvdputte** [comment]
@bnomei I updated the plugin with the loglevels. Check it out.

@iskrisis Indeed, static function is probably not the best idea. I decided to go with a helper function. The namespacing part in the options is now how you suppose to do it with K3-plugins AFAIK?
